nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/utils/AzureStorageUtils.java

```diff
@@ -85,6 +86,20 @@
             .sensitive(true)
             .build();
 
+    public static final PropertyDescriptor ENDPOINT_SUFFIX = new PropertyDescriptor.Builder()
+            .name("storage-endpoint-suffix")
+            .displayName("Common Storage Account Endpoint Suffix")
+            .description(
+                    "Storage accounts in public Azure always use a common FQDN suffix. " +
+                    "Override this endpoint suffix with a different suffix in certain circumsances (like Azure Stack or non-public Azure regions). " +
+                    "The preferred way is to configure them through a controller service specified in the Storage Credentials property. " +
+                    "The controller service can provide a common/shared configuration for multiple/all Azure processors. Furthermore, the credentials " +
+                    "can also be looked up dynamically with the 'Lookup' version of the service.")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .required(false)
```

Comment:
If we don't use `allowableValues()` this will need a validator.

Reply:
Null default value is acceptable. The Azure Storage SDK sets the default end suffix, if we pass null. Do we still need a validator in this case? I validated the regression behavior without setting end suffix property.

Reply:
In order to support Azure  Stack on-premise, this should be an editable free form.

Reply:
@sjyang18 Re: Azure Stack endpoint, sounds good. Thanks for digging into it.

Re: validation. I think it'd be helpful to account for the just whitespace or _Set empty string_ scenarios to help prevent mistakes.

Reply:
I added NON_BLANK_VALIDATOR.